src/Symfony/Component/Security/Http/RememberMe/RememberMeHandlerInterface.php

```diff
@@ -51,4 +52,25 @@ public function consumeRememberMeCookie(RememberMeDetails $rememberMeDetails): U
      * This should set a cookie with a `null` value on the request attribute.
      */
     public function clearRememberMeCookie(): void;
+
+    /**
+     * Retrieves the User Identifier for the RememberMe cookie.
+     *
+     * If the cookie is required to contain the User Identifier {@see UserInterface::getUserIdentifier()}
+     * then it can simply be retrieved from the provided cookie details; if not, then it could be retrieved
+     * from elsewhere based on some other information provided (e.g. in a database).
+     *
+     * Cannot be added to 6.1 because of backwards compatibility.
+     */
+    //public function getUserIdentifierForCookie(RememberMeDetails $rememberMeDetails): string;
+
+    /**
+     * Retrieves the RememberMeDetails using the raw cookie.
+     *
+     * This method allows the authenticator to retrieve the cookie details without needing
+     * to care about the implementation details of the RememberMeDetails used by the RememberMeHandler.
+     *
+     * Cannot be added to 6.1 because of backwards compatibility.
+     */
+    //public function getRememberMeDetails(string $rawCookie): RememberMeDetails;
```

Comment:
The new methods need to be added as virtual methods using the `@method` annotation. See e.g. https://github.com/symfony/symfony/blob/143ad6a04bbd7e8e73e35afb3d4c2438886273a2/src/Symfony/Component/PasswordHasher/Hasher/UserPasswordHasherInterface.php#L21

Reply:
I was already wondering how to handle these methods. Changes have been made.